knora/dsplib/utils/xml_upload.py

```diff
@@ -910,94 +918,340 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
+                if pure_text not in hash_to_value:
+                    continue
+                new_xmltext = hash_to_value[pure_text]
+
+                # replace the outdated internal ids by their IRI
+                for _id, _iri in id2iri_mapping.items():
+                    new_xmltext.regex_replace(f'href="IRI:{_id}:IRI"', f'href="{_iri}"')
+
+                # prepare API call
+                jsonobj = {
+                    "@id": res_iri,
+                    "@type": resource.restype,
+                    link_prop.name: {
+                        "@id": existing_value['@id'],
+                        "@type": "knora-api:TextValue",
+                        "knora-api:textValueAsXml": new_xmltext,
+                        "knora-api:textValueHasMapping": {
+                            '@id': 'http://rdfh.ch/standoff/mappings/StandardMapping'
+                        }
+                    },
+                    "@context": existing_resource['@context']
+                }
+                jsondata = json.dumps(jsonobj, indent=4, separators=(',', ': '), cls=KnoraStandoffXmlEncoder)
+
+                # execute API call
+                answer = None
+                for _ in range(5):
+                    try:
+                        answer = con.put(path='/v2/values', jsondata=jsondata)
+                        stashed_xml_texts[resource][link_prop].pop(pure_text)
+                        break
+                    except ConnectionError:
+                        print(f'{datetime.now().isoformat()}: Try reconnecting to DSP server...')
+                        time.sleep(1)
+                        continue
+                    except RequestException:
+                        print(f'{datetime.now().isoformat()}: Try reconnecting to DSP server...')
+                        time.sleep(1)
+                        continue
+                    except BaseError:
+                        print(f'ERROR while updating the xml text of "{link_prop.name}" of resource "{resource.id}"')
+                        break
+                if verbose:
+                    if answer:
+                        print(f'  Successfully updated xml text of "{link_prop.name}"\n')
 
-    # update the resources with the stashed resptrs
-    if len(stashed_resptr_props) > 0:
-        print('Update the stashed resptrs...')
-    for resource, prop_2_resptrs in stashed_resptr_props.items():
+    # make a purged version of stashed_xml_texts, without empty entries
+    nonapplied_xml_texts: dict[XMLResource, dict[XMLProperty, dict[str, KnoraStandoffXml]]] = {}
+    for res, propdict in stashed_xml_texts.items():
+        for prop, xmldict in propdict.items():
+            if len(xmldict) > 0:
+                if res not in nonapplied_xml_texts:
+                    nonapplied_xml_texts[res] = {}
+                nonapplied_xml_texts[res][prop] = xmldict
+
+    return nonapplied_xml_texts
+
+
+def update_stashed_resptr_props(
```

Comment:
At first glance the name of this method seems misleading to me because the stashed resptr props are not updated, instead the resources are updated with the resptr props. Or am I mistaken?